### DISCUSSION TITLE
Game refuses to start (render error)

### DISCUSSION THREAD

**Discussion by znepb** [cc-tweaked/CC-Tweaked#1547] [Q&A] [answered]
### Minecraft Version

1.20.x

### Version

1.106.1 (fabric)

### Details

I hope this log is helpful enough. I just created an instance in IntelIJ to make a new mod and it just crashes when CCT is in my mods folder. See crash report here: https://mclo.gs/2QG8r9r and log here: https://mclo.gs/6sLzP8Z

**Comment by SquidDev** [comment, marked as answer]
Fabric 0.86 deprecated one of the modules that we use, which means it's no longer present in a dev environment by default. We've updated to the replacement API in dev (24d74f5c80ff051ed0d695de4a446eeeb21af454), but that hasn't been released yet.

As a workaround, you can add the following dependency in Gradle:

```groovy
modRuntimeOnly("net.fabricmc.fabric-api:fabric-api-deprecated:${project.fabric_version}")
```

**Comment by znepb** [comment]
I tried putting that in my build.gradle, but now Fabric gives me this on startup:
```
Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
A potential solution has been determined:
	 - Replace mod 'Fabric Model Loading API (v1)' (fabric-model-loading-api-v1) 1.0.2+709a987177 with any version that is compatible with:
		 - fabric-models-v0 0.3.35+b3afc78b77
Unmet dependency listing:
	 - Mod 'Fabric Model Loading API (v1)' (fabric-model-loading-api-v1) 1.0.2+709a987177 is incompatible with any version before 0.4.0 of mod 'Fabric Models (v0)' (fabric-models-v0), but a matching version is present: 0.3.35+b3afc78b77!
	at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:51)
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:192)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	at net.fabricmc.devlaunchinjector.Main.main(Main.java:86)
```
It seems as though the new version of the modeling api is incompatible with the old version but I don't know how I can exclude this from Gradle. Any ideas?

**Reply by SquidDev** [reply]
Can you post your build.gradle and gradle.properties?

**Reply by znepb** [reply]
build.gradle: https://gist.github.com/znepb/c631f3a899881df7e00993b4fdba390b
gradle.properties: https://gist.github.com/znepb/730e69c4b788d98ecfa15c41b60d3901

**Reply by znepb** [reply]
So, adding CC into my run/mods folder causes the game to crash, but with the mod built and put into a normal Minecraft instance with CC:T, the game runs fine.

**Reply by SquidDev** [reply]
Thank you! That's very odd - I've just set up a workspace locally, and the game (and CC!) loads fine for me.

> adding CC into my run/mods folder causes the game to crash

Oh, you definitely don't want to do that! CC:T should already be loaded (as it's on the classpath), and so you don't need to put it in `run/mods` again.

**Reply by znepb** [reply]
Even after nuking my .gradle and build folders the problems persists. Hopefully it just fixes itself

**Reply by znepb** [reply]
Kinda starting to think this is a Fabric issue, I'll make an issue over there

**Reply by znepb** [reply]
After some thinking, I decided that building CC 1.20.1 myself and importing the jar file into my gradle might work, and it did! But now, CC's child mods aren't loading, which causes a NoClassDefFoundError when trying to access the config API. I'll try adding all the required mods to my libs folder along with CC, but if there's a better way to do this please let me know.

**Reply by Lupus590** [reply]
What are you trying to do by building the mod yourself? Why are you adding mods to the run folder?

**Reply by znepb** [reply]
The most recent commit of the mod has been updated to use the latest Fabric loader, while the maven has not. I'm not adding anything to the run folder, I'm just including dependencies.
